### PR TITLE
Draft: Reduce Page Size With One Optimized SVG File

### PR DESCRIPTION
**Pull request by Ctannerweb**

### CODE REVIEW

File: source/posts/2015-04-27-svg-optimization.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+Feel free to do any kind of styling to the SVG.
+
+## What have we done?
+We have taken the best of two techniques.
+
+The inline HTML is clean and simple while only using one .svg file so the browser does not have to make multiple trips back to the server.
+
+As [Brian Cardarella](https://twitter.com/bcardarella) explains it.
+
+> Think of images on your website as books. Would you rather carry one book over to the bookshelf or have a bunch of books that require multiple trips to the bookshelf.
```
Comment (Ctannerweb) [inline source/posts/2015-04-27-svg-optimization.md]: @bcardarella I recalled Amanda telling me this is how you explained it. If you want it re-worded let me know :+1:

Comment (bcardarella) [inline source/posts/2015-04-27-svg-optimization.md]: I said something similar to this. I think I was using pens as an example.

File: source/posts/2015-04-27-svg-optimization.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+social: true
+author: Cory Tanner
+twitter: "ctannerweb"
+ithub: ctannerweb
+summary: "Showing you how to combine multiple svg files into a single optimized SVG and then use it site wide."
+published: true
+tags: html, css
+---
+
+When it comes to SVG’s it can be hard to decide on the best practice for implementing multiple SVG’s (Scalable Vector Graphics) throughout your website is.
```
Comment (poteto) [inline source/posts/2015-04-27-svg-optimization.md]: This sentence seems a little clumsy. 

Better:

> It's sometimes hard to decide on best practices for implementing multiple SVGs (Scalable Vector Graphics) in your app or website.

File: source/posts/2015-04-27-svg-optimization.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+Currently at [DockYard](https://twitter.com/DockYard "DockYard Twitter") we are working on a new project which includes rethinking graphic optimization. The obvious choice is to have every graphic possible in a SVG format rather than .png files.
```
Comment (poteto) [inline source/posts/2015-04-27-svg-optimization.md]: Be consistent: `SVG` and `PNG`, or `.svg` and `.png`

File: source/posts/2015-04-27-svg-optimization.md (hunk 2 above)
Comment (Ctannerweb) [inline source/posts/2015-04-27-svg-optimization.md]: Ya i'm not happy with my intro yet... Working on a better hook.

File: source/posts/2015-04-27-svg-optimization.md (hunk 1 above)
Comment (Ctannerweb) [inline source/posts/2015-04-27-svg-optimization.md]: @bcardarella Ok, just wanted to make sure you are ok with it.

File: source/posts/2015-04-27-svg-optimization.md [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+summary: "Showing you how to combine multiple svg files into a single optimized SVG and then use it site wide."
+published: true
+tags: html, css
+---
+
+It's sometimes hard to decide on best practices for implementing multiple SVGs (Scalable Vector Graphics) in your app or website.
+
+Currently at [DockYard](https://twitter.com/DockYard "DockYard Twitter") we are working on a new project which includes rethinking graphic optimization. The obvious choice is to have every graphic possible in a SVG format rather than .png files.
+
+With the now mandatory responsive design standards for websites you need vector images that scale cleanly on any screen size.
```
Comment (acacheung) [inline source/posts/2015-04-27-svg-optimization.md]: Maybe link out to other articles that show why SVG's are a good choice if viable

File: source/posts/2015-04-27-svg-optimization.md [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+
+SVG’s are the obvious choice but how to organize them is the real question.
+
+# Lets get into the fun stuff
+There are two main techniques of implementing SVG’s into your website.
```
Comment (acacheung) [inline source/posts/2015-04-27-svg-optimization.md]: I just came across this too http://tobyj.net/responsive-sprites/
thoughts?

Comment (acacheung) [inline source/posts/2015-04-27-svg-optimization.md]: Maybe "Before I thought there were two main techniques..."

File: source/posts/2015-04-27-svg-optimization.md (hunk 4 above)
Comment (poteto) [inline source/posts/2015-04-27-svg-optimization.md]: And a link to the article that mentions how Google now has responsive design as a metric for pagerank

File: source/posts/2015-04-27-svg-optimization.md [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+tags: html, css
+---
+
+It's sometimes hard to decide on best practices for implementing multiple SVGs (Scalable Vector Graphics) in your app or website.
+
+Currently at [DockYard](https://twitter.com/DockYard "DockYard Twitter") we are working on a new project which includes rethinking graphic optimization. The obvious choice is to have every graphic possible in a SVG format rather than .png files.
+
+With the now mandatory responsive design standards for websites you need vector images that scale cleanly on any screen size.
+
+SVG’s are the obvious choice but how to organize them is the real question.
```
Comment (poteto) [inline source/posts/2015-04-27-svg-optimization.md]: ~~SVG's~~ SVGs

`SVG's === SVG is`

File: source/posts/2015-04-27-svg-optimization.md [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+
+# Lets get into the fun stuff
+There are two main techniques of implementing SVG’s into your website.
+
+Placing `<svg>` code into your HTML directly (inline SVG).
```
Comment (poteto) [inline source/posts/2015-04-27-svg-optimization.md]: Use an ordered list

File: source/posts/2015-04-27-svg-optimization.md [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+
+Save a .svg file to your `/image` folder for every vector image and style a `<div>` with the background-image as the .svg file
```
Comment (poteto) [inline source/posts/2015-04-27-svg-optimization.md]: Use an ordered list

File: source/posts/2015-04-27-svg-optimization.md [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+
+Both have advantages and disadvantages that I want to go over before I present the solution.
+
+### Inline SVG
+With a inline SVG you are doing just how it sounds, copying SVG code directly into your HTML.
```
Comment (poteto) [inline source/posts/2015-04-27-svg-optimization.md]: This line is redundant, `inline SVG` is quite self-explanatory. The main point here is to describe *why* you would prefer an inline SVG over an external asset.

File: source/posts/2015-04-27-svg-optimization.md [hunk 10]
```diff
@@ -0,0 +1,198 @@
...
+</svg>
+```
+
+Editing the `<path>` is not normally necessary but you should edit the `<svg>` and add a class. The important thing to keep in mind is that with the class you can style your `<svg>` in your css.
```
Comment (poteto) [inline source/posts/2015-04-27-svg-optimization.md]: `CSS` not css

File: source/posts/2015-04-27-svg-optimization.md (hunk 4 above)
Comment (Ctannerweb) [inline source/posts/2015-04-27-svg-optimization.md]: @poteto is this the one you are referring to? http://googlewebmastercentral.blogspot.com/2015/02/finding-more-mobile-friendly-search.html

File: source/posts/2015-04-27-svg-optimization.md [hunk 11]
```diff
@@ -0,0 +1,198 @@
...
+
+The positives with Inline SVG is that it is extremely lightweight. Something this small would only weigh about 2kb. That size is dependent on the complexity of the paths inside the SVG tags.
```
Comment (poteto) [inline source/posts/2015-04-27-svg-optimization.md]: IMO the positive is not file size (an SVG would probably be a larger file size as HTML vs SVG) but the elimination of another `GET` request for the SVG. With an inline SVG, the graphic is loaded together with the HTML and is rendered immediately once the HTML loads. With an external SVG, the browser has to issue a `GET` request to that resource, download it, and then render it.

File: source/posts/2015-04-27-svg-optimization.md [hunk 12]
```diff
@@ -0,0 +1,198 @@
...
+
+When you start to get more complicated SVG’s with groups and multiple paths your code length will begin to increase greatly. You can end up with over 100 lines of SVG code in your HTML if your graphic is too complex.
```
Comment (poteto) [inline source/posts/2015-04-27-svg-optimization.md]: You mention the code length increase, And the downside for anything inline (e.g. CSS, JS) is mainly that of maintainability

File: source/posts/2015-04-27-svg-optimization.md [hunk 13]
```diff
@@ -0,0 +1,198 @@
...
+![](http://i.imgur.com/GfiKWKd.png "Project")
+
+If we make independent .svg files and then need to add more in the future we will start to affect page load times with all the files the browser has to pull from the server.
+
+The solution is to make one file the browser has get but apply inline SVG code that asks for only a certain group ID inside the .svg that we will specify. This technique will only show the graph we want even when there are 3 graphs in one .svg file.
+
+### Making your .SVG File
+In illustrator make sure the artboard width and height are equal to all three graphs width and height when stacked on top of each other. In this example we have 360px by 108px (width/height) graphs.
+
+**Step One**
```
Comment (poteto) [inline source/posts/2015-04-27-svg-optimization.md]: Use an ordered list instead

**Pull request closed without merging**